src/main/resources/lib/categorizedview/categorized-view.js

```diff
@@ -0,0 +1,5 @@
+window.addEventListener("DOMContentLoaded", function() {
```

Comment:
This isn't a direct extraction but rather an extraction and a modification (wrapping in a `DOMContentLoaded` listener), isn't it? What is the reason for the modification?

Reply:
This is a mistake - it should be `document` instead of `window`. 

It still works without waiting for the entire DOM to be loaded, are you suggesting to just use 

```
document.querySelectorAll("table.categorizedSortable").forEach(function (e){
        e.sortable = new CategorizedSortable.CategorizedSortable(e);
    })
```

Reply:
I am not suggesting anything in particular. I am asking what is the reason you have modified the original script when extracting it.

Reply:
In this case, we want to make sure `table.categorizedSortable` is loaded before trying to get the element from the DOM. Similar to #75.

Reply:
> In this case, we want to make sure `table.categorizedSortable` loaded before trying to get the elements from the DOM.

Is that not ensured in the existing code by virtue of the `<script>` tag being after the `<table>` tag? Is there any reason to believe that would not remain the case when extracting the inline `<script>` to a separate file without any modifications?

Reply:
In some cases it might not matter (like this one when script is included on the bottom of the page). But it's not guaranteed that every script is included on the bottom and that it will only execute when we have whole DOM loaded. So this is a good practice to follow. It is also recommended(-ish) by https://www.jenkins.io/doc/developer/security/csp/#inline-event-handlers.

Reply:
As a general rule, it may make sense to put `<script>` tags in the `<head>` section and use `DOMContentLoaded`, but that is orthogonal to the CSP fix in this PR.

Reply:
Thanks for clarifying this. I made the relevant change and will keep this in mind moving forward.

Reply:
I don't think it should've been removed :/
While it doesn't affect anything in this particular case it may not work in other cases.

Reply:
Using the event handler is inconsistent with putting the `<script>` tag at the end of the document. It is fine to move the `<script>` tag to the `<head>` section and to add the event handler, but this was a separate code cleanup and not needed for the CSP compatibility fix in this PR.